cpp/src/community/louvain.cu

```diff
@@ -87,8 +87,20 @@ std::pair<size_t, weight_t> louvain(
       runner(handle, graph_view);
 
     weight_t wt = runner(max_level, resolution);
-    // TODO: implement this...
-    // runner.get_dendrogram().partition_at_level(clustering, runner.get_dendrogram().num_levels());
+
+    rmm::device_uvector<vertex_t> vertex_ids_v(graph_view.get_number_of_vertices(),
+                                               handle.get_stream());
+
+    thrust::copy(rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
+                 thrust::make_counting_iterator<vertex_t>(0),
+                 thrust::make_counting_iterator<vertex_t>(graph_view.get_number_of_vertices()),
+                 vertex_ids_v.begin());
+
+    partition_at_level<vertex_t, multi_gpu>(handle,
+                                            runner.get_dendrogram(),
+                                            vertex_ids_v.data(),
+                                            clustering,
+                                            runner.get_dendrogram().num_levels());
 
     // FIXME: Consider returning the Dendrogram at some point
```

Comment:
I could consider moving the FIXME, but I think it is still relevant.

The current python API (and every version since we integrated) expects the C++ code to flatten the result and return the final clustering.  The networkx python implementation of this returns a Dendrogram and the caller can choose at what level to flatten the Dendrogram later.  This seems better... but I didn't want to break the python API at this point.